README.md

```diff
@@ -0,0 +1,144 @@
+# LeanCode TypeScript Contracts Generator
+
+## Available plugins
+
+- [contracts](plugins/contracts/) - type generation
+- [client](plugins/client/) - api factory generation; requires `contracts` to also be used
+- [raw](plugins/raw) - prepend/append text to generated file;
+- [admin](plugins/admin) - api components schema generation; used by [Admin App Generator](packages/api-admin/)
+
+## Usage
+
+```sh
+# you also have to install all plugins used by your config e.g.
+# @leancodepl/contractsgenerator-typescript-plugin-contracts @leancodepl/contractsgenerator-typescript-plugin-client @leancodepl/contractsgenerator-typescript-plugin-raw
+npm i -D @leancodepl/contractsgenerator-typescript
+
+npx @leancodepl/contractsgenerator-typescript
+```
+
+One thing to remember is that TypeScript Contracts Generator relies on
+[Contracts Generator Server](https://github.com/leancodepl/contractsgenerator). Consequently Contracts Generator Server
+needs `dotnet` as its runtime. That means in order for generator to work you need to have `dotnet` runtime installed.
+Another thing which also needs to be noted - currently TypeScript Contracts Generator works in POSIX shells (due to
+`generate.sh` script). This means **on Windows you need to be using Git Bash or alternative**.
+
+### Configuration
+
+Contracts generator is configured using [lilconfig](https://github.com/antonk52/lilconfig). Valid configuration sources
+include:
+
+- `contractsgenerator-typescript` property in `package.json`,
+- `.contractsgenerator-typescriptrc.json` for raw JSON
+- `contractsgenerator-typescript.config.js`, `contractsgenerator-typescript.config.cjs`,
+  `.contractsgenerator-typescriptrc.js`, `.contractsgenerator-typescriptrc.cjs` for configuration using JavaScript.
+  Those files need to export the configuration object.
+- path to JavaScript/JSON/YAML config file passed via `--config/-c` parameter
+
+### Config structure
+
+- `generates`**\*** - dictionary of files to generate where key is path to generated file and value is file
+  configuration. File configuration consists of `plugins` field and optional `config` field
+  - `plugins`**\*** - list of plugins that will be used to generate this file. Plugin is specified either by it's name
+    or dictionary with single field whose key is plugin name and value is it's config which replaces other already
```

Comment:
This sentence is so convoluted that I had hard time understanding it even when knowing how it works :sweat_smile: